src/main/java/be/seeseemelk/mockbukkit/inventory/meta/SkullMetaMock.java

```diff
@@ -77,27 +81,36 @@ public boolean equals(Object obj)
 			return false;
 		}
 
-		return Objects.equals(owner, other.getOwningPlayer().getName());
+		return Objects.equals(playerProfile.getName(), other.getOwningPlayer().getName());
 	}
 
 	@Override
 	@Deprecated(since = "1.13")
 	public String getOwner()
 	{
-		return owner;
+		return hasOwner() ? playerProfile.getName() : null;
 	}
 
 	@Override
 	public boolean hasOwner()
 	{
-		return !Strings.isNullOrEmpty(owner);
+		return playerProfile != null && !playerProfile.getName().isEmpty();
 	}
 
 	@Override
 	@Deprecated(since = "1.13")
-	public boolean setOwner(String owner)
+	public boolean setOwner(String name)
 	{
-		this.owner = owner;
+		if (name != null && name.length() > MAX_OWNER_LENGTH) {
+			return false;
+		}
+
+		if (name == null) {
+			setPlayerProfile(null);
+		} else {
+			setPlayerProfile(new PlayerProfileMock(name, UUID.randomUUID()));
```

Comment:
Should firstly look for registered players matching said name as before generating an uuid. The uuid should also be `new UUID(0L, 0L);` , code looks like this in paper:
```java
 else {
            // Paper start - Use Online Players Skull
            GameProfile newProfile = null;
            net.minecraft.server.level.ServerPlayer player = net.minecraft.server.MinecraftServer.getServer().getPlayerList().getPlayerByName(name);
            if (player != null) newProfile = player.getGameProfile();
            if (newProfile == null) newProfile = new GameProfile(Util.NIL_UUID, name);
            this.setProfile(newProfile);
            // Paper end
        }
```

Reply:
Ah okay I will tweak this tomorrow I was basing everything of bukkit code instead of paper

Reply:
Please base future Contributions on Paper Code. A guide on how to aquire it can be found in [Paper's Contributing Guidelines](https://github.com/PaperMC/Paper/blob/master/CONTRIBUTING.md)

Reply:
There were questions around implementing gameprofile because it can do network requests.
We need a decission on if we want to implement gameprofile or not.

Reply:
You can't just use ServerMock.getPlayer(String)?

I think it's fine to just check the locally registered players.